src/pathfinding_igvc/src/PathToTwistNode.cpp

```diff
@@ -0,0 +1,155 @@
+
+PathToTwistNode::PathToTwistNode(int argc, char** argv, std::string node_name) {
+    // Setup NodeHandles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+    uint32_t queue_size = 1;
+
+    /*No received coordinates from tf on launch so not valid */
+    valid_cood = false;
+
+    std::string path_subscribe_topic = "/path"; // Setup subscriber to path
+    path_subscriber                  = nh.subscribe(
+    path_subscribe_topic, queue_size, &PathToTwistNode::pathCallBack, this);
+
+    std::string tf_subscribe_topic = "/tf"; // Setup subscriber to tf
+    tf_subscriber                  = nh.subscribe(
+    tf_subscribe_topic, queue_size, &PathToTwistNode::tfCallBack, this);
+
+    std::string twist_publish_topic =
+    private_nh.resolveName("/cmd_vel"); // setup Publisher to twist
+    twist_publisher =
+    nh.advertise<geometry_msgs::Twist>(twist_publish_topic, queue_size);
+
+    // Get Params
+    SB_getParam(
+    private_nh, "base_frame", base_frame, (std::string) "base_link");
+    SB_getParam(
+    private_nh, "global_frame", global_frame, (std::string) "odom_combined");
+    SB_getParam(private_nh, "num_poses", num_poses, 10);
+}
+
+void PathToTwistNode::pathCallBack(const nav_msgs::Path::ConstPtr& path_ptr) {
+    nav_msgs::Path path_msg =
+    *path_ptr; // Take required information from received message
+    geometry_msgs::Twist twist_msg = pathToTwist(path_msg,
+                                                 robot_x_pos,
+                                                 robot_y_pos,
+                                                 robot_orientation,
+                                                 num_poses,
+                                                 valid_cood);
+    twist_publisher.publish(twist_msg);
+}
+
+void PathToTwistNode::tfCallBack(const tf2_msgs::TFMessageConstPtr tf_message) {
+    // Check if the tf_message contains the transform we're looking for
+    for (geometry_msgs::TransformStamped tf_stamped : tf_message->transforms) {
+        if (tf_stamped.header.frame_id == global_frame &&
+            tf_stamped.child_frame_id == base_frame) {
+            double x_pos = tf_stamped.transform.translation.x;
+            double y_pos = tf_stamped.transform.translation.y;
+
+            double quatx = tf_stamped.transform.rotation.x;
+            double quaty = tf_stamped.transform.rotation.y;
+            double quatz = tf_stamped.transform.rotation.z;
+            double quatw = tf_stamped.transform.rotation.w;
+            tf::Quaternion quat(quatx, quaty, quatz, quatw);
+            tf::Matrix3x3 rot_matrix(quat);
+            double roll, pitch, yaw;
+            rot_matrix.getRPY(roll, pitch, yaw);
+
+            // Set member variables
+            robot_x_pos = x_pos;
+            robot_y_pos = y_pos;
+            robot_orientation =
+            yaw; // Orientation = rotation about z axis (yaw)
+            valid_cood = true;
```

Comment:
what is valid_cood? does it stand for valid coordinate?

Reply:
sorry, everything under `pathfinding_igvc` was intended for another PR, I've removed it all for this one.